src/app/components/common-ui/WebForm/index.js

```diff
@@ -0,0 +1,335 @@
+import _ from 'underscore';
+import cn from 'classnames';
+import op from 'object-path';
+import PropTypes from 'prop-types';
+
+import React, {
+    forwardRef,
+    useEffect,
+    useLayoutEffect as useWindowEffect,
+    useImperativeHandle,
+    useRef,
+    useState,
+} from 'react';
+
+const useLayoutEffect =
+    typeof window !== 'undefined' ? useWindowEffect : useEffect;
+
+const noop = () => {};
+
+const ENUMS = {
+    STATUS: {
+        COMPLETE: 'COMPLETE',
+        READY: 'READY',
+        SENDING: 'SENDING',
+        ERROR: 'ERROR',
+    },
+};
+
+/**
+ * -----------------------------------------------------------------------------
+ * Hook Component: WebForm
+ * -----------------------------------------------------------------------------
+ */
+let WebForm = (props, ref) => {
+    const {
+        className,
+        namespace,
+        required,
+        onBeforeSubmit,
+        onComplete,
+        onError,
+        onSubmit,
+        onUpdate,
+        showError,
+        validator,
+        value,
+        name,
+        id,
+        children,
+        ...formProps
+    } = props;
+
+    // Refs
+    const formRef = useRef();
+    const stateRef = useRef({
+        value,
+        error: null,
+        mounted: false,
+        status: ENUMS.STATUS.READY,
+        elements: {},
+    });
+
+    // State
+    const [, setNewState] = useState(stateRef.current);
```

Comment:
This is new to me. I've never seen this syntax. I'm guessing it allows you to trigger render w/o really saving anything to memory?

Reply:
Yeah, you can put any number of commas into the destructuring and skip those elements of the array.